app/lib/three_scale/api/collection.rb

```diff
@@ -1,42 +1,67 @@
-class ThreeScale::Api::Collection
-
-  include PaginationHelper::ClassMethods
+# frozen_string_literal: true
 
+class ThreeScale::Api::Collection
   def initialize(collection, options = {})
     @collection = collection
     @options = options
   end
 
-  def to_xml(options = {})
-    builder = options[:builder] || ThreeScale::XML::Builder.new
+  def to_xml(xml_options = {})
+    builder = xml_options[:builder] || ThreeScale::XML::Builder.new
+    options.merge!(xml_options).merge!(builder: builder, skip_instruct: true)
```

Comment:
Kind of weird that `#to_xml` mutates the state of the object.

Reply:
makes sense 👍🏼